src/_macosx.m

```diff
@@ -737,7 +825,10 @@ static CGFloat _get_device_scale(CGContextRef cr)
     rect.origin.y = 350;
     rect.size.height = height;
     rect.size.width = width;
-
+    if (_set_icon() != 0)
```

Comment:
Why don't you do this in the python-side `FigureManagerMac.__init__` (just like for qt)? then you can get rid of the c-level _get_pdf_icon_path.

Reply:
Good idea. I didn’t know that was an option. Should clean things up a bit.